certification/pyxis/submit.go

```diff
@@ -17,31 +20,42 @@ func (p *pyxisClient) SubmitResults(ctx context.Context, certInput *certificatio
 	certImage := certInput.CertImage
 
 	// Submission effectively starts the certification process, so switch
-	// the status to reflect this if needed.
+	// the status to reflect this if needed. This only needs to be done for net new projects.
+	// Existing projects that are in "In Progress" can stay "In Progress" until they moved to "Published" which is triggered
+	// once an image in a project is moved to "Published" status. The status on the project would stay in "Published" status,
+	// unless the partner decides to un-publish all of their images. At that point backed systems/processes would move
+	// the project back to "In Process" and there would still be nothing that preflight need to update on the project.
 	if certProject.CertificationStatus == "Started" {
 		certProject.CertificationStatus = "In Progress"
-	}
 
-	// You must have an existing repository.
-	if len(certImage.Repositories) == 0 {
-		return nil, errors.ErrInvalidCertImage
-	}
+		// You must have an existing repository.
+		if len(certImage.Repositories) == 0 {
+			return nil, errors.ErrInvalidCertImage
+		}
 
-	// Set this project's metadata to match the image that we're certifying.
-	certProject.Container.Registry = certImage.Repositories[0].Registry
-	certProject.Container.Repository = certImage.Repositories[0].Repository
+		// Setting registry to the value we get from certImage from crane and then normalizing
+		// index.docker.io to docker.io so project info shows properly in the Red Hat Catalog
+		registry := certImage.Repositories[0].Registry
+		if registry == name.DefaultRegistry {
+			registry = defaultRegistryAlias
+		}
```

Comment:
We never update the project after the first image is submitted. Since this value cannot be changed on the project page in the UI, we need to normalize that upfront. The value in the UI at the project level drives what ends up in the catalog and for people to `pull` the image via this value. So it only makes sense to normalize here and not everwhere in our code since we still ned `index.docker.io` in the `crane` values so we can still do authenticated calls for say `list-tags` or anything else that would require auth.

TL:DR only net new cert projects are updated by preflight.

Reply:
I'm not following; I'll let someone else review here.

Reply:
Spoke with @acornett21.

This bit changes the registry listing for effectively new projects to `docker.io` if crane internally parsed the image's registry to live in Docker Hub. 

Part of the issue for me is that this value doesn't get set to what the user provided. It's definitely coming from Docker hub, so in practice, it's mostly no harm and no foul. It just seemed strange.

Secondly, this doesn't change the image registry value to what the user provided when submitting to pyxis. Again, I would say it's somewhat fair to assume that the value we give pyxis (which I'm fairly certain is always index.docker.io) should function as expected, and so changing the image value back to what the user gave us may not net us benefit. It just feels inconsistent.

Either way, functionally, I see where this is going. Seems like it should work.

I'll leave this unresolved for folks to review to see if it seemed odd or if it was just me (entirely possible, as I don't know if I understand the submit flow that well).